x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlAuthenticator.java

```diff
@@ -395,4 +402,30 @@ private void checkLifetimeRestrictions(Conditions conditions) {
     private void checkLifetimeRestrictions(SubjectConfirmationData subjectConfirmationData) {
         validateNotOnOrAfter(subjectConfirmationData.getNotOnOrAfter());
     }
+
+    private void warnOnSpecialAttributeNames(Assertion assertion, List<Attribute> attributes) {
+        attributes.forEach(attribute -> {
+            warnOnSpecialAttributeName(assertion, attribute.getName(), "name");
+
+            String attributeFriendlyName = attribute.getFriendlyName();
+            if (attributeFriendlyName != null) {
+                warnOnSpecialAttributeName(assertion, attributeFriendlyName, "friendly name");
+            }
+        });
+    }
+
+    private void warnOnSpecialAttributeName(Assertion assertion, String attributeName, String fieldNameForLogMessage) {
+        if (SPECIAL_ATTRIBUTE_NAMES.contains(attributeName)) {
+            logger.warn(
+                "SAML assertion [{}] has attribute with {} [{}] which clashes with a special attribute name. "
+                    + "Attributes with a name clash cannot be mapped. "
+                    + "Change attribute {} to not clash with any of [{}].",
```

Comment:
I know you've done 1 round on this message already, but I think it's worth another go.

The issue is, in the scenario we're dealing with, we can assume that the administrator doesn't really understand SAML very well (because they made this mistake) and they really need an error message that helps them without assuming much knowledge.

```suggestion
                "SAML assertion [{}] has attribute with {} [{}] which clashes with a special attribute name. "
                    + "Attributes with a name clash may prevent authentication or interfere will role mapping. "
                    + "Change your IdP configuration to use a different attribute {}"
                    + " that will not clash with any of [{}]" 
```

Reply:
Perfect, this one is way clearer! Thanks for the suggestion.